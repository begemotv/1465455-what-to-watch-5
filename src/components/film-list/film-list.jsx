import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import FilmCard from "../film-card/film-card";
import LoadMoreButton from "../load-more-button/load-more-button";
import {changeActiveFilmId, changeActiveFilmIdGenre} from "../../store/action";
import {getFilmsByGenre} from "../../store/selectors";
import {filmPropTypes} from "../../prop-types";

const FilmList = (props) => {
  const {
    activeItem,
    filmsByGenre,
    moviesShownOverall,
    changeActiveFilmAction,
    onItemInteraction,
    onItemInteractionEnd,
    onShowMoreButtonClick
  } = props;

  return (
    <React.Fragment>
      <div className="catalog__movies-list">
        {(filmsByGenre.slice(0, moviesShownOverall))
            .map((film) =>
              <FilmCard
                key={film.id}
                film={film}
                onCardClick={() => changeActiveFilmAction(film.id, film.genre)}
                onItemInteraction={onItemInteraction}
                onItemInteractionEnd={onItemInteractionEnd}
                isCardActive={film.id === activeItem}
              />)}
      </div>
      {moviesShownOverall <= filmsByGenre.length &&
            <LoadMoreButton onShowMoreButtonClick={onShowMoreButtonClick}/>}
    </React.Fragment>
  );
};

FilmList.propTypes = {
  activeItem: PropTypes.number.isRequired,
  changeActiveFilmAction: PropTypes.func.isRequired,
  filmsByGenre: PropTypes.arrayOf(
      PropTypes.shape(filmPropTypes)
  ).isRequired,
  moviesShownOverall: PropTypes.number.isRequired,
  onItemInteraction: PropTypes.func.isRequired,
  onItemInteractionEnd: PropTypes.func.isRequired,
  onShowMoreButtonClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  filmsByGenre: getFilmsByGenre(state),
});

const mapDispatchToProps = (dispatch) => ({
  changeActiveFilmAction(id, genre) {
    dispatch(changeActiveFilmId(id));
    dispatch(changeActiveFilmIdGenre(genre));
  },
});

export {FilmList};
export default connect(mapStateToProps, mapDispatchToProps)(FilmList);
