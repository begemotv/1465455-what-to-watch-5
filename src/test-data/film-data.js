export const filmMock = {
  backgroundColor: `#73B39A`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 1,
  isFavorite: false,
  name: `Bronson`,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  rating: 3.6,
  ratingName: `Normal`,
  releaseYear: 2008,
  runningTime: `1h 32m`,
  starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  voteCount: 1635992,
};

export const filmsMock = [
  {
    backgroundColor: `#73B39A`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    isFavorite: false,
    name: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    rating: 3.6,
    ratingName: `Normal`,
    releaseYear: 2008,
    runningTime: `1h 32m`,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 1635992,
  },
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 2,
    isFavorite: false,
    name: `Pulp Fiction`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    rating: 1.5,
    ratingName: `Bad`,
    releaseYear: 1994,
    runningTime: `2h 33m`,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 1635992,
  },
  {
    backgroundColor: `#828585`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
    description: `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
    director: `Jeff Nichols`,
    genre: `Action`,
    id: 3,
    isFavorite: false,
    name: `Midnight Special`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
    rating: 3.3,
    ratingName: `Normal`,
    releaseYear: 2016,
    runningTime: `1h 52m`,
    starring: [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
    videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoSrcTrailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    voteCount: 67815,
  },
  {
    backgroundColor: `#B9B27E`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    description: `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    director: `Wachowski Brothers`,
    genre: `Action`,
    id: 4,
    isFavorite: false,
    name: `Matrix`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    rating: 4.4,
    ratingName: `Normal`,
    releaseYear: 1999,
    runningTime: `2h 16m`,
    starring: [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 1503092,
  },
  {
    backgroundColor: `#E1DAD7`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
    description: `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
    director: `Brian Helgeland`,
    genre: `Crime`,
    id: 5,
    isFavorite: false,
    name: `Legend`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
    rating: 3.5,
    ratingName: `Normal`,
    releaseYear: 2015,
    runningTime: `2h 12m`,
    starring: [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
    videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoSrcTrailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    voteCount: 138487,
  },
  {
    backgroundColor: `#A6B7AC`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/gangs_of_new_york.jpg`,
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    director: `Martin Scorsese`,
    genre: `Crime`,
    id: 6,
    isFavorite: false,
    name: `Gangs of new york`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    rating: 8.8,
    ratingName: `Very good`,
    releaseYear: 2002,
    runningTime: `2h 47m`,
    starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    voteCount: 370881,
  },
  {
    backgroundColor: `#C4C0C0`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/A_Star_is_Born.jpg`,
    description: `A musician helps a young singer find fame as age and alcoholism send his own career into a downward spiral.`,
    director: `Bradley Cooper`,
    genre: `Drama`,
    id: 7,
    isFavorite: false,
    name: `A Star Is Born`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_Is_Born.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg`,
    rating: 3.9,
    ratingName: `Normal`,
    releaseYear: 2018,
    runningTime: `2h 16m`,
    starring: [`Lady Gaga`, `Bradley Cooper`, `Sam Elliott`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 244484,
  },
  {
    backgroundColor: `#EBC996`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/beach.jpg`,
    description: `Vicenarian Richard travels to Thailand and finds himself in possession of a strange map. Rumours state that it leads to a solitary beach paradise, a tropical bliss. Excited and intrigued, he sets out to find it.`,
    director: `Danny Boyle`,
    genre: `Adventure`,
    id: 8,
    isFavorite: false,
    name: `Beach`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/beach.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg`,
    rating: 3.3,
    ratingName: `Normal`,
    releaseYear: 2000,
    runningTime: `1h 59m`,
    starring: [`Leonardo DiCaprio`, `Daniel York`, `Patcharawan Patarakijjanon`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    voteCount: 207824,
  },
  {
    backgroundColor: `#D6CDAF`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Aviator.jpg`,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    director: `Martin Scorsese`,
    genre: `Drama`,
    id: 9,
    isFavorite: false,
    name: `Aviator`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Aviator.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg`,
    rating: 9.8,
    ratingName: `Very good`,
    releaseYear: 2014,
    runningTime: `2h 50m`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    videoSrc: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoSrcTrailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    voteCount: 307174,
  },
  {
    backgroundColor: `#F0DBA2`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Johnny_English.jpg`,
    description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
    director: `Peter Howitt`,
    genre: `Comedy`,
    id: 10,
    isFavorite: false,
    name: `Johnny English`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Johnny_English.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg`,
    rating: 10,
    ratingName: `Awesome`,
    releaseYear: 2003,
    runningTime: `1h 28m`,
    starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 136843,
  },
  {
    backgroundColor: `#AD9F8B`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Dardjeeling_Limited.jpg`,
    description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
    director: `Wes Anderson`,
    genre: `Adventure`,
    id: 11,
    isFavorite: false,
    name: `Dardjeeling Limited`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Dardjeeling_Limited.jpg`,
    previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/dardjeeling_limited.jpg`,
    rating: 3.6,
    ratingName: `Normal`,
    releaseYear: 2007,
    runningTime: `1h 31m`,
    starring: [`Owen Wilson`, `Adrien Brody`, `Jason Schwartzman`],
    videoSrc: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoSrcTrailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    voteCount: 165106,
  }
];

export const reviewsMock = [
  {
    id: 1,
    rating: 7.5,
    text: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    timeStamp: `2020-10-08T13:38:44.769Z`,
    user: {
      id: 17,
      name: `Emely`
    },
  },
  {
    id: 2,
    rating: 5.8,
    text: `I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.`,
    timeStamp: `2020-11-06T13:38:44.769Z`,
    user: {
      id: 17,
      name: `Emely`
    },
  },
  {
    id: 3,
    rating: 7.9,
    text: `I really find it difficult to believe this movie is rated highly by people. It's hands down the worst movie I've seen in my life`,
    timeStamp: `2020-11-04T13:38:44.769Z`,
    user: {
      id: 10,
      name: `Max`},
  }
];
