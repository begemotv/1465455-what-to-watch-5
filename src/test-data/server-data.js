export const filmMockServer = {
  "background_color": `#B9B27E`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "director": `Wachowski Brothers`,
  "genre": `Action`,
  "id": 4,
  "is_favorite": false,
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  "rating": 4.4,
  "released": 1999,
  "run_time": 136,
  "scores_count": 1503092,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
};

export const filmMockFavoriteServer = {
  "background_color": `#B9B27E`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
  "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
  "director": `Wachowski Brothers`,
  "genre": `Action`,
  "id": 4,
  "is_favorite": true,
  "name": `Matrix`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
  "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  "rating": 4.4,
  "released": 1999,
  "run_time": 136,
  "scores_count": 1503092,
  "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
};

export const filmsMockServer = [
  {
    "background_color": `#B9B27E`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    "director": `Wachowski Brothers`,
    "genre": `Action`,
    "id": 4,
    "is_favorite": false,
    "name": `Matrix`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    "rating": 4.4,
    "released": 1999,
    "run_time": 136,
    "scores_count": 1503092,
    "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
  {
    "background_color": `#795433`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "director": `Quentin Tarantino`,
    "genre": `Crime`,
    "id": 2,
    "is_favorite": false,
    "name": `Pulp Fiction`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    "rating": 1.5,
    "released": 1994,
    "run_time": 153,
    "scores_count": 1635992,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  },
  {
    "background_color": `#BDAD8F`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    "description": `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    "director": `Ethan Coen`,
    "genre": `Crime`,
    "id": 14,
    "is_favorite": false,
    "name": `No Country for Old Men`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    "rating": 4.1,
    "released": 2007,
    "run_time": 122,
    "scores_count": 764976,
    "starring": [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  }
];

export const userMockServer = {
  "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg`,
  "email": `1@fdfs.com`,
  "id": 1,
  "name": `Bob`,
};

export const reviewsMockServer = [
  {
    "comment": `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    "date": `2020-11-03T13:38:44.769Z`,
    "id": 1,
    "rating": 2.4,
    "user": {
      "id": 11,
      "name": `Jack`
    },
  },
  {
    "comment": `A movie that will take you to another world full of emotions.`,
    "date": `2020-10-23T13:38:44.769Z`,
    "id": 2,
    "rating": 7.2,
    "user": {
      "id": 12,
      "name": `Isaac`
    },
  }
];
