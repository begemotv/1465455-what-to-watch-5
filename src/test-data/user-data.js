export const userMock = {
  avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg`,
  email: `1@fdfs.com`,
  id: 1,
  name: `1`,
};
