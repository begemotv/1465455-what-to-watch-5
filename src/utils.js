export const nameRating = (rating) => {
  let ratingName = ``;
  switch (true) {
    case rating >= 0 && rating < 3:
      ratingName = `Bad`;
      break;
    case rating >= 3 && rating < 5:
      ratingName = `Normal`;
      break;
    case rating >= 5 && rating < 8:
      ratingName = `Good`;
      break;
    case rating >= 8 && rating < 10:
      ratingName = `Very good`;
      break;
    case rating === 10:
      ratingName = `Awesome`;
      break;
    default:
      ratingName = `Not rated`;
  }
  return ratingName;
};

