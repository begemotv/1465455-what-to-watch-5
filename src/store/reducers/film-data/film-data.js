import {extend} from "../../../utils";
import {ActionType} from "../../action";

const initialState = {
  activeFilmDetails: null,
  activeFilmDetailsPromo: null,
  activeFilmReviews: [],
  films: [],
  isFetching: true,
  isReviewFormBlocked: false,
};

export const filmData = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FILMS:
      return extend(state, {
        films: action.payload,
      });
    case ActionType.LOAD_ACTIVE_FILM_DETAILS:
      return extend(state, {
        activeFilmDetails: action.payload,
      });
    case ActionType.LOAD_ACTIVE_FILM_DETAILS_PROMO:
      return extend(state, {
        activeFilmDetailsPromo: action.payload,
      });
    case ActionType.LOAD_ACTIVE_FILM_REVIEWS:
      return extend(state, {
        activeFilmReviews: action.payload,
      });
    case ActionType.CHANGE_REVIEW_FORM_STATUS:
      return extend(state, {
        isReviewFormBlocked: action.payload,
      });
    default:
      return state;
  }
};
