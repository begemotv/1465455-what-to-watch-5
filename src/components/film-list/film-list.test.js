import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";

import {FilmList} from "./film-list";
import {noop} from "../../utils";
import {filmsMock} from "../../test-data/film-data";

const activeItemMock = 1;

test(`FilmList renders correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <FilmList
            activeItem={activeItemMock}
            films={filmsMock}
            changeActiveFilmAction={noop}
            onItemInteraction={noop}
            onItemInteractionEnd={noop}
          />
        </MemoryRouter>
        , {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
