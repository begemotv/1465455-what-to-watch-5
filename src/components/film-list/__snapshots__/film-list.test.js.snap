// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Bronson"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Bronson
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/2"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Pulp Fiction"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Pulp Fiction
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/3"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Midnight Special"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Midnight Special
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/4"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Matrix"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Matrix
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/5"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Legend"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Legend
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/6"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Gangs of new york"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Gangs of new york
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/7"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="A Star Is Born"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        A Star Is Born
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/8"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Beach"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Beach
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/9"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Aviator"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Aviator
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/10"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Johnny English"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Johnny English
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/11"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Dardjeeling Limited"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/dardjeeling_limited.jpg"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Dardjeeling Limited
      </h3>
    </a>
  </article>
</div>
`;
